l2geth/rollup/sync_service.go

```diff
@@ -763,10 +680,211 @@ func (s *SyncService) ApplyTransaction(tx *types.Transaction) error {
 		txRaw,
 	)
 	tx.SetTransactionMeta(newMeta)
-
 	return s.applyTransaction(tx)
 }
 
+// syncTransactionsToTip will sync all of the transactions to the tip.
+// The Backend determines the source of the transactions.
+func (s *SyncService) syncTransactionsToTip(backend Backend) error {
+	s.loopLock.Lock()
+	defer s.loopLock.Unlock()
+
+	for {
+		latest, err := s.client.GetLatestTransaction(backend)
+		if errors.Is(err, errElementNotFound) {
+			log.Info("No transactions to sync")
+			return nil
+		}
+		if err != nil {
+			return fmt.Errorf("Cannot get latest transaction: %w", err)
+		}
+		latestIndex := latest.GetMeta().Index
+		if latestIndex == nil {
+			return errors.New("Latest index is nil")
+		}
+
+		nextIndex := s.GetNextIndex()
+		if nextIndex-1 == *latestIndex {
+			log.Info("No new transactions to sync", "tip", *latestIndex, "backend", backend.String())
+			return nil
+		}
+		if err := s.syncTransactionRange(nextIndex, *latestIndex, backend); err != nil {
+			return err
+		}
+
+		post, err := s.client.GetLatestTransaction(backend)
+		if err != nil {
+			return fmt.Errorf("Cannot get latest transaction: %w", err)
+		}
+		postLatestIndex := post.GetMeta().Index
+		if postLatestIndex == nil {
+			return errors.New("Latest index is nil")
+		}
+
+		if *postLatestIndex == *latestIndex {
+			return nil
+		}
+	}
+}
+
+// syncTransactionRange will sync a range of transactions from
+// start to end (inclusive) from a specific Backend
```

Comment:
The `syncXXXRange` functions can be updated to be concurrent with their HTTP requests in the future, low hanging fruit optimization